xtask/codegen/src/generate_configuration.rs

```diff
@@ -91,21 +91,27 @@ pub(crate) fn generate_rules_configuration(mode: Mode) -> Result<()> {
             #property_group_name: None
         });
 
-        let global_recommended = if group == "nursery" {
-            quote! { self.is_recommended() && biome_flags::is_unstable() }
+        let (global_all, global_recommended) = if group == "nursery" {
+            (
+                quote! { self.is_all() && biome_flags::is_unstable() },
+                quote! { self.is_recommended() && biome_flags::is_unstable() },
+            )
         } else {
-            quote! { self.is_recommended() }
+            (quote! { self.is_all() }, quote! { self.is_recommended() })
         };
 
         group_as_default_rules.push(quote! {
             if let Some(group) = self.#property_group_name.as_ref() {
-                group.collect_preset_rules(self.is_recommended(), &mut enabled_rules, &mut disabled_rules);
+                group.collect_preset_rules(
+                    #global_all,
+                    #global_recommended,
+                    &mut enabled_rules,
+                    &mut disabled_rules,
+                );
                 enabled_rules.extend(&group.get_enabled_rules());
                 disabled_rules.extend(&group.get_disabled_rules());
-            } else if self.is_all() {
+            } else if #global_all {
                 enabled_rules.extend(#group_struct_name::all_rules_as_filters());
-            } else if self.is_not_all() {
-                disabled_rules.extend(#group_struct_name::all_rules_as_filters());
```

Comment:
Could you justify why this lines are deleted?

If `all` is explicitly set to `false`, we should certainly add the rules to `disabled_rules`?

Reply:
These rules are not in `enabled_rules`, so we don't have to put them in `disabled_rules`.

Reply:
Thanks for the clarification. LGTM!